tests/system/python/conftest.py

```diff
@@ -410,6 +410,294 @@ def _read_data_from_pi(host, admin, password, pi_database, asset, sensor):
+            res = conn.getresponse()
+            assert res.status == 200, "Databases is not accessible."
+            r = json.loads(res.read().decode())
+            for el in r["Items"]:
+                if el["Name"] == pi_database:
+                    url_elements_list = el["Links"]["Elements"]
+
+        print("Getting old cache")
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/system/cacheinstances', headers=normal_header)
+        res = conn.getresponse()
+        r = json.loads(res.read().decode())
+        try:
+            # assuming we have single user Administrator
+            old_cache_refresh_time = r["Items"][0]['LastRefreshTime']
+        except (IndexError, KeyError):
+            print("The cache does not exist. ")
+            return
+
+        print("Old cache refresh time {} ".format(old_cache_refresh_time))
+        conn.close()
+
+        print("Going to request element list with cache control no cache.")
+        if url_elements_list is not None:
+            conn.request("GET", url_elements_list, headers=headers)
+            res = conn.getresponse()
+            r = json.loads(res.read().decode())
+
+        conn.close()
+
+        # for verification whether we are able to clear cache.
+        print("Now verifying whether cache cleared.")
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/system/cacheinstances', headers=normal_header)
+        res = conn.getresponse()
+        r = json.loads(res.read().decode())
+        try:
+            # assuming we have only one user named Administrator.
+            new_cache_refresh_time = r["Items"][0]['LastRefreshTime']
+        except (KeyError, IndexError):
+            print("The cache does not exist.")
+            return
+
+        print("New cache refresh time {} ".format(new_cache_refresh_time))
+        conn.close()
+
+        assert new_cache_refresh_time != old_cache_refresh_time, "The cache has not been refreshed."
+
+    except Exception as er:
+        print("Could not clear cache due to {}".format(er))
+        assert False, "Could not clear cache due to {}".format(er)
+
+
+@pytest.fixture
+def clear_pi_system_through_pi_web_api():
+    def _clear_pi_system_through_pi_web_api(host, admin, password, pi_database, af_hierarchy_list, asset_dict):
+        """
+           Clears the pi system through pi web API.
+           1. Deletes the elements.
+           2. Deletes element templates.
+           3. Deletes the PI Points.
+           4. Clears the cache.
+        Args:
+            host (str): The address of the pi server.
+            admin (str): The user name inside pi server.
+            password (str): The passowrd for the username used above.
+            pi_database (str): The database inside pi server.
+            af_hierarchy_list (list): The asset heirarchy list to delete.
```

Comment:
Typo

Reply:
Ok I will open a new pull request for that.